tests/Functional/EndToEndTest.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace PhpTuf\ComposerStager\Tests\Functional;
+
+use Symfony\Component\Process\Process;
+
+/**
+ * @coversNothing
+ */
+class EndToEndTest extends TestCase
+{
+    public static function setUpBeforeClass(): void
+    {
+        self::createTestEnvironment();
+        self::initializeComposerJson();
+    }
+
+    public static function tearDownAfterClass(): void
+    {
+        self::removeTestEnvironment();
+    }
+
+    protected static function runFrontScript(array $args, string $cwd = __DIR__): Process
```

Comment:
Practically all the supporting methods in tests, including assertions, have to be static for consistency or because they have to be callable by static methods like `::tearDownAfterClass()` that inherit directly from PHPUnit.